### PR TITLE
net/mdns-repeater: Modernize plugin code; Add blocklist support (contributed by Kodehyrden)

### PR DESCRIPTION
**Pull request by Monviech**
A general cleanup of the plugin code because I was tracing a bug in the view with tokenized fields but it caused a chain reaction that resulted in this PR.

This includes: https://github.com/opnsense/plugins/pull/4362

### CODE REVIEW

File: net/mdns-repeater/src/etc/inc/plugins.inc.d/mdnsrepeater.inc [hunk 1]
```diff
@@ -1,63 +1,65 @@
...
-    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-    POSSIBILITY OF SUCH DAMAGE.
-*/
-
-function mdnsrepeater_enabled()
-{
-    $model = new \OPNsense\MDNSRepeater\MDNSRepeater();
-    return (string)$model->enabled == '1';
-}
+ * Copyright © 2017 Fabian Franz
```
Comment (fichtner) [inline net/mdns-repeater/src/etc/inc/plugins.inc.d/mdnsrepeater.inc:4]:
```suggestion
 * Copyright (c) 2017 Fabian Franz
```

File: net/mdns-repeater/src/opnsense/mvc/app/models/OPNsense/MDNSRepeater/MDNSRepeater.xml [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
+        <enablecarp type="BooleanField">
+            <Default>0</Default>
+            <Required>Y</Required>
+        </enablecarp>
+        <interfaces type="InterfaceField">
+            <Default>lan</Default>
+            <Required>Y</Required>
+            <Multiple>Y</Multiple>
+        </interfaces>
+        <blacklist type="NetworkField">
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/mvc/app/models/OPNsense/MDNSRepeater/MDNSRepeater.xml:19]: let's please call this "blocklist"

File: net/mdns-repeater/src/opnsense/mvc/app/views/OPNsense/MDNSRepeater/index.volt [hunk 3]
```diff
@@ -1,63 +1,75 @@
...
-AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
-OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
-SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
-CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-POSSIBILITY OF SUCH DAMAGE.
-
-#}
+ # Copyright © 2017 Fabian Franz
```
Comment (fichtner) [inline net/mdns-repeater/src/opnsense/mvc/app/views/OPNsense/MDNSRepeater/index.volt:2]:
```suggestion
 # Copyright (c) 2017 Fabian Franz
```

File: net/mdns-repeater/pkg-descr [hunk 4]
```diff
@@ -7,6 +7,11 @@ It can be used to bridge zeroconf devices to work properly across the two subnet
 Plugin Changelog
 ================
 
+1.2
+
+* modernize plugin code
+* blacklist support (contributed by Kodehyrden)
```
Comment (fichtner) [inline net/mdns-repeater/pkg-descr:13]:
```suggestion
* blocklists support (contributed by Kodehyrden)
```

File: net/mdns-repeater/src/opnsense/mvc/app/models/OPNsense/MDNSRepeater/MDNSRepeater.xml (hunk 2 above)
Comment (Monviech) [reply net/mdns-repeater/src/opnsense/mvc/app/models/OPNsense/MDNSRepeater/MDNSRepeater.xml:19]: But its called blacklist in source:

https://github.com/geekman/mdns-repeater/releases/tag/1.11

Comment (fichtner) [reply net/mdns-repeater/src/opnsense/mvc/app/models/OPNsense/MDNSRepeater/MDNSRepeater.xml:19]: Not our problem. 😊

Comment (Monviech) [reply net/mdns-repeater/src/opnsense/mvc/app/models/OPNsense/MDNSRepeater/MDNSRepeater.xml]: @fichtner The deed has been done.

**Review by fichtner — APPROVED** [review]

**Pull request merged**